packages/cli/src/PublicApi/v1/Credentials/credentials.handler.ts

```diff
@@ -0,0 +1,90 @@
+import express = require('express');
+
+import { CredentialsEntity } from '../../../databases/entities/CredentialsEntity';
+import { CredentialRequest } from '../../../requests';
+import { externalHooks } from '../../../Server';
+
+import {
+	createCredential,
+	encryptCredential,
+	getCredentials,
+	getSharedCredentials,
+	removeCredential,
+	sanitizeCredentials,
+	saveCredential,
+} from './credentials.service';
+
+export = {
+	createCredential: [
+		async (
+			req: CredentialRequest.Create,
+			res: express.Response,
+		): Promise<express.Response<Partial<CredentialsEntity>>> => {
+			delete req.body.id; // delete if sent
+
+			try {
+				const newCredential = await createCredential(req.body as Partial<CredentialsEntity>);
+
+				const encryptedData = await encryptCredential(newCredential);
+
+				Object.assign(newCredential, encryptedData);
+
+				await externalHooks.run('credentials.create', [encryptedData]);
+
+				const savedCredential = await saveCredential(newCredential, req.user);
+
+				// LoggerProxy.verbose('New credential created', {
+				// 	credentialId: newCredential.id,
+				// 	ownerId: req.user.id,
+				// });
+
+				return res.json(sanitizeCredentials(savedCredential));
+			} catch ({ message, httpStatusCode }) {
+				httpStatusCode ? res.status(httpStatusCode) : res.status(500);
+				return res.json({ message });
+			}
+		},
+	],
+	deleteCredential: [
+		async (
+			req: CredentialRequest.Delete,
+			res: express.Response,
+		): Promise<express.Response<Partial<CredentialsEntity>>> => {
+			const { id: credentialId } = req.params;
+			let credentials: CredentialsEntity | undefined;
+
+			if (req.user.globalRole.name !== 'owner') {
+				const shared = await getSharedCredentials(req.user.id, credentialId, [
+					'credentials',
+					'role',
+				]);
+
+				if (shared && shared.role.name !== 'owner') {
```

Comment:
The only way you can be in the share table is with the owner role, so I guess this condition is not needed.

Reply:
at the moment this is true, it will change in a few weeks though ^^ question is if we already put it in or leave it out?

Reply:
We can leave it for now. But whenever you change user management, we will probably have to update all endpoints to take this into account.